### PR TITLE
Add more tests to ValueTranslatorSpec

### PR DESCRIPTION
**Pull request by paulbrauner-da**
Also found and fixed a minor bug in `PackageInterface.lookupEnumConstructor`.

### CODE REVIEW

File: sdk/daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/ValueTranslatorSpec.scala [hunk 1]
```diff
@@ -429,29 +496,65 @@ class ValueTranslatorSpec
...
+            typ shouldBe TRecordUpgradable
+          },
+        ),
         (
           TRecordUpgradable,
           ValueRecord(
             "",
             ImmArray(
               "fieldA" -> aInt,
-              "fieldB" -> someParty, // Here the field has type Party instead of Text
```
Comment (dylant-da) [inline sdk/daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/ValueTranslatorSpec.scala:438]: Is there a reason we're dropping this test of field mismatches to create a test of an extra non-optional field? Couldn't we make the non-optional field test a wholly new test instead?

File: sdk/daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/ValueTranslatorSpec.scala [hunk 2]
```diff
@@ -429,29 +496,65 @@ class ValueTranslatorSpec
...
+          },
+        ),
+        (
+          TEnumUpgradable,
+          ValueEnum("", "Cons3"), // Cons3 is not a constructor of Mod:Enum
+          {
+            case Error.Preprocessing.Lookup(
+                  LookupError.NotFound(
+                    Reference.DataEnumConstructor(_, consName),
+                    _,
```
Comment (dylant-da) [inline sdk/daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/ValueTranslatorSpec.scala:552]: Can we assert that this wildcard has the shape `Reference.DataEnumConstructor(_, _)`?

**Review by dylant-da — APPROVED** [review]
LGTM

File: sdk/daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/ValueTranslatorSpec.scala (hunk 1 above)
Comment (paulbrauner-da) [reply sdk/daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/ValueTranslatorSpec.scala:438]: This test case was an accidental duplicate of the same test case on line 403.

**Pull request merged**